src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -128,6 +133,63 @@ public bool TryRead(out T value)
             return true;
         }
 
+        /// <summary>
+        /// Move the reader back the specified number of positions.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Retreat(int count)
```

Comment:
Rewind is a better name I think- I'm putting that name in the proposal.

Reply:
Also, can you make this 'long'?

Reply:
Yep. Made the count in Advance long, should follow suit.